R/add_lines.R

```diff
@@ -0,0 +1,25 @@
+#' Add lines
+#'
+#' @param data dataset to divide into lines
+#' @param extract_length total length of extract in ms
+#' @param window_size length of line in ms
+#'
+#' @return data set with 'line' column added
+#' @export
+#'
+add_lines <- function(data,extract_length=600000,window_size=60000) {
+
+  window_breaks <- as.integer(c(0:round(extract_length/window_size)) * window_size)
+
+  data <- data |>
+    dplyr::mutate(end = end - min(begin), # reset timestamps to start from 0
+           begin = begin - min(begin),
```

Comment:
I don't like that timestamps are adjusted here without communication to the user. This should be done in a different column that gets deleted later, or not at all.

Reply:
excellent point — and not necessary actually. A clever plotting function should be able to deal with convos that happen to start at the 5min mark (and should indeed preserve that — also because otherwise we break the link to audio